File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<int> fillValues(int rows){
        int ans = 1;
        vector<int> temp;
        temp.push_back(1);

        for(int i = 1; i < rows; i++){
            ans = ans * (rows - i);
            ans = ans / i;
            temp.push_back(ans);
        }
        return temp;
    }

    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        for(int i = 1; i <= numRows; i++){
            ans.push_back(fillValues(i));
        }
        return ans;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def helper(self, rows):
        ans = 1
        temp = []
        temp.append(ans)

        for i in range(1, rows):
            ans = ans * (rows - i)
            ans //= i
            temp.append(ans)
        
        return temp

    def generate(self, numRows: int) -> List[List[int]]:
        ans = []
        for i in range(1, numRows+1):
            ans.append(self.helper(i))
        
        return ans